lib/twom.h

```diff
@@ -0,0 +1,162 @@
+/* twom.h - twoskip implementation with MVCC capability
+ *
+ * Copyright (c) 2025 Fastmail Pty Ltd
+ *
+ * https://creativecommons.org/publicdomain/zero/1.0/
+ *
+ *   The person who associated a work with this deed has dedicated the work to the
+ *   public domain by waiving all of his or her rights to the work worldwide under
+ *   copyright law, including all related and neighboring rights, to the extent
+ *   allowed by law.
+ *
+ *   You can copy, modify, distribute and perform the work, even for commercial
+ *   purposes, all without asking permission.
+ */
+
+#ifndef INCLUDED_TWOM_H
+#define INCLUDED_TWOM_H
+
+#include <stdlib.h>
+#include <stdint.h>
+
+struct twom_db;
+struct twom_txn;
+struct twom_cursor;
+
+enum twom_ret {
```

Comment:
Do these enums have to match the ones from CYRUSDB? `_errormap` in cyrusdb_twom.c converts them to CYRUSDB errors anyway. If these enums still better match exactly the CYRUSDB error values, state it in a comment here. The missing -7 and -8 look particularly odd otherwise.

Reply:
done, alphabetised as well